comp/remote-config/rcclient/rcclientimpl/rcclient.go

```diff
@@ -291,14 +294,26 @@ func (rc rcClient) agentConfigUpdateCallback(updates map[string]state.RawConfig,
 
 		// Need to update the log level even if the level stays the same because we need to update the source
 		// Might be possible to add a check in deeper functions to avoid unnecessary work
-		err = rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC)
+		if err := rc.settingsComponent.SetRuntimeSetting("log_level", mergedConfig.LogLevel, model.SourceRC); err != nil {
+			errs = append(errs, err)
+		}
+	}
+
+	streamLogsSource := config.Datadog().GetSource("logs_config.streaming.enable_streamlogs")
+	if mergedConfig.EnableStreamLogs {
+		if err := rc.settingsComponent.SetRuntimeSetting("enable_streamlogs", mergedConfig.EnableStreamLogs, model.SourceRC); err != nil {
+			errs = append(errs, err)
+		}
+	} else if !mergedConfig.EnableStreamLogs && streamLogsSource == model.SourceRC {
+		config.Datadog().UnsetForSource("logs_config.streaming.enable_streamlogs", model.SourceRC)
 	}
 
 	// Apply the new status to all configs
 	for cfgPath := range updates {
-		if err == nil {
+		if errs == nil {
```

Comment:
A slice can be non-nil but still empty. Not with the current code, but checking length would be more future-proof:

```suggestion
		if len(errs) == 0 {
```